Source/EbsdLib/IO/AngleFileLoader.cpp

```diff
@@ -0,0 +1,349 @@
+
+  // Make sure the file exists on disk
+  if(!fi.exists())
+  {
+    setErrorMessage("Input File does not exist at path");
+    setErrorCode(-2);
+    return angles;
+  }
+
+  // Make sure we have a valid angle representation
+  if(m_AngleRepresentation != EulerAngles && m_AngleRepresentation != QuaternionAngles && m_AngleRepresentation != RodriguezAngles)
+  {
+    setErrorMessage("The Angle representation was not set to anything known to this code");
+    setErrorCode(-3);
+    return angles;
+  }
+
+  // The format of the file is quite simple. Comment lines start with a "#" symbol
+  // The only Key-Value pair we are looking for is 'Angle Count' which will have
+  // the total number of angles that will be read
+
+  int numOrients = 0;
+  QByteArray buf;
+
+  // Open the file and read the first line
+  QFile reader(getInputFile());
+  if(!reader.open(QIODevice::ReadOnly | QIODevice::Text))
+  {
+    QString msg = QObject::tr("Angle file could not be opened: %1").arg(getInputFile());
+    setErrorCode(-100);
+    setErrorMessage(msg);
+    return angles;
+  }
+
+  bool ok = false;
+  buf = reader.readLine();
+  while(buf[0] == '#')
+  {
+    buf = reader.readLine();
+  }
+  buf = buf.trimmed();
+
+  // Split the next line into a pair of tokens delimited by the ":" character
+  QList<QByteArray> tokens = buf.split(':');
+  if(tokens.count() != 2)
+  {
+    QString msg = QObject::tr("Proper Header was not detected. The file should have a single header line of 'Angle Count:XXXX'");
+    setErrorCode(-101);
+    setErrorMessage(msg);
+    return angles;
+  }
+
+  if(tokens[0].toStdString() != "Angle Count")
+  {
+    QString msg = QObject::tr("Proper Header was not detected. The file should have a single header line of 'Angle Count:XXXX'");
+    setErrorCode(-102);
+    setErrorMessage(msg);
+    return angles;
+  }
+  numOrients = tokens[1].toInt(&ok, 10);
+
+  // Allocate enough for the angles
+  std::vector<size_t> dims(1, 5);
+  angles = EbsdLib::FloatArrayType::CreateArray(numOrients, dims, "EulerAngles_From_File", true);
+
+  for(int i = 0; i < numOrients; i++)
+  {
+    float weight = 0.0f;
+    float sigma = 1.0f;
+    buf = reader.readLine();
+    // Skip any lines that start with a '#' character
+    if(buf[0] == '#')
+    {
+      continue;
+    }
+    buf = buf.trimmed();
+
+    // Remove multiple Delimiters if wanted by the user.
+    if(m_IgnoreMultipleDelimiters)
+    {
+      buf = buf.simplified();
+    }
+    QString delimiter = getDelimiter();
+    if(delimiter.compare("\t") == 0)
+    {
+      setDelimiter(" ");
+    }
+    tokens = buf.split(*(getDelimiter().toLatin1().data()));
+
+    OrientationF euler(3);
+    if(m_AngleRepresentation == EulerAngles)
+    {
+      euler[0] = tokens[0].trimmed().toFloat(&ok);
+      euler[1] = tokens[1].trimmed().toFloat(&ok);
+      euler[2] = tokens[2].trimmed().toFloat(&ok);
+      weight = tokens[3].trimmed().toFloat(&ok);
+      sigma = tokens[4].trimmed().toFloat(&ok);
```

Comment:
If you aren't checking ok, use the default parameter value (nullptr).
This applies to later uses as well as this method has a number of them.